### PR TITLE
fix: No way to go back to appsmith applications from the app viewer mode navigation

### LINKED ISSUE #13854: [Bug]: No way to go back to appsmith applications from the app viewer mode navigation
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

1. As part of theming change to the navigation, we now don't have a way to navigate to appsmith apps
2. A user who clicks on comment mode cannot come out of it unless they press escape

### Steps To Reproduce

Appsmith apps
1. Deploy the app
2. Observe the appsmith logo is not available in the navigation
3. A user doesn't have a way to go to their apps

Comment mode
1. Click on comment in the view mode
2. Click the button again but it doesn't come out of comment mode

### Public Sample App

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by jsartisan**
This PR adds back to home button in header in view mode

## Description

Fixes #13854

## Type of change
- Bug fix

## How Has This Been Tested?
- Added cypress tests

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Parthvi12 — COMMENTED** [review]
@jsartisan Profile.snap.png should match the display name and email which is there in CI i.e. testowner@appsmith.com otherwise this will fail.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/ViewMode/ViewMode_spec.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  });
+
+  it("on click of apps on header, it should take to application home page", function() {
+    cy.PublishtheApp();
+
+    cy.get(".t--back-to-home").click();
+    cy.url().should("eq", Cypress.config().baseUrl + "applications");
+  });
+
+  afterEach(() => {
```
Comment (keyurparalkar) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/ViewMode/ViewMode_spec.js:17]: Can we remove this?

File: app/client/src/pages/AppViewer/BackToHomeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { ANONYMOUS_USERNAME } from "constants/userConstants";
+import { getSelectedAppTheme } from "selectors/appThemingSelectors";
+
+function BackToHomeButton() {
+  const history = useHistory();
+  const currentUser = useSelector(getCurrentUser);
+  const selectedTheme = useSelector(getSelectedAppTheme);
+
+  // if user is not logged in, don't render anything
+  if (currentUser?.username == ANONYMOUS_USERNAME) return null;
```
Comment (keyurparalkar) [inline app/client/src/pages/AppViewer/BackToHomeButton.tsx:16]: Is it possible to move this logic to it's parent component? i.e. in `AppViewerHeader`

**Review by keyurparalkar — APPROVED** [review]

**Comment by keyurparalkar** [comment]
LGTM!

File: app/client/src/pages/AppViewer/AppViewerHeader.tsx [hunk 3]
```diff
@@ -167,15 +105,20 @@ export function AppViewerHeader(props: AppViewerHeaderProps) {
                   <MenuIcon className="w-5 h-5" />
                 )}
               </div>
-              <div className="w-full ml-2 text-center md:text-left md:ml-0">
-                <div className="text-base font-medium text-gray-600">
-                  {currentApplicationDetails?.name}
-                </div>
+              <div className="">
+                {currentUser?.username == ANONYMOUS_USERNAME && (
```
Comment (aswathkk) [inline app/client/src/pages/AppViewer/AppViewerHeader.tsx:109]: Why this is enabled only for Anonymous users?

**Review by aswathkk — COMMENTED** [review]
@jsartisan 
Changes look good. But, I have a few queries:
1. Why are we enabling the Back to home button only for anonymous users?
2. Why the issue related to Comment mode is not implemented in this PR?

File: app/client/src/pages/AppViewer/AppViewerHeader.tsx (hunk 3 above)
Comment (aswathkk) [reply app/client/src/pages/AppViewer/AppViewerHeader.tsx:109]: Did you mean `!==` ?

**Comment by jsartisan** [comment]
> Why are we enabling the Back to home button only for anonymous users?
That was a typo on my end. Fixed it.

> Why the issue related to Comment mode is not implemented in this PR?
The commenting feature will be removed altogether in a few weeks. So we are not spending time on any bugs related to commenting feature.

**Review by aswathkk — APPROVED** [review]
LGTM

**Pull request merged**